pkg/services/ppmshipment/rules_test.go

```
@@ -476,4 +476,44 @@ func (suite *PPMShipmentSuite) TestValidationRules() {
 			suite.Error(err)
 		})
 	})
+
+	suite.Run("CheckEstimatedWeight()", func() {
+		suite.Run("success estimatedWeight set", func() {
+			shipmentID := uuid.Must(uuid.NewV4())
+			estimatedWeight := unit.Pound(4000)
+			estimatedIncentive := int32(17000)
+
+			oldPPMShipment := models.PPMShipment{
+				ShipmentID:      shipmentID,
+				EstimatedWeight: &estimatedWeight,
+			}
+
+			newPPMShipment := models.PPMShipment{
+				ShipmentID:         shipmentID,
+				EstimatedWeight:    &estimatedWeight,
+				EstimatedIncentive: &estimatedIncentive,
+			}
+
+			err := checkEstimatedWeight().Validate(suite.AppContextForTest(), newPPMShipment, &oldPPMShipment, nil)
+			suite.NilOrNoVerrs(err)
+		})
+
+		suite.Run("failure estimatedWeight cannot be nil", func() {
+			shipmentID := uuid.Must(uuid.NewV4())
+			estimatedWeight := unit.Pound(4000)
+
+			oldPPMShipment := models.PPMShipment{
+				ShipmentID:      shipmentID,
+				EstimatedWeight: nil,
+			}
+
+			newPPMShipment := models.PPMShipment{
+				ShipmentID:      shipmentID,
+				EstimatedWeight: &estimatedWeight,
+			}
+
+			err := checkEstimatedWeight().Validate(suite.AppContextForTest(), newPPMShipment, &oldPPMShipment, nil)
+			suite.Error(err)
```

Your validation looks right but this test is silently succeeding because verrs is not nil but is empty.  The newPPMShipment should have EstimatedWeight set to nil.

In this case you could change your test to make sure the error message matches:
`suite.Equal("cannot be empty", err.Error())`